addons/l10n_in_edi/models/account_move.py

```diff
@@ -225,7 +230,37 @@ def _l10n_in_edi_send_invoice(self):
                         "doc_date": self.invoice_date and self.invoice_date.strftime("%d/%m/%Y"),
                     }
                 )
-                if not response.get("error"):
+                mismatch_error = []
+                decoded_response = {}
+                if jwt:
+                    try:
+                        if signedinvoice := response.get('data', {}).get('SignedInvoice', {}):
+                            decoded_response = jwt.decode(signedinvoice, options={'verify_signature': False})
+                            decoded_response = json.loads(decoded_response.get('data', '{}'))
+                    except (json.JSONDecodeError, jwt.exceptions.DecodeError) as e:
+                        _logger.warning("Failed to decode SignedInvoice JWT payload: %s", str(e))
+                if decoded_response:
+                    received_gstin = decoded_response.get('BuyerDtls', {}).get('Gstin')
+                    expected_gstin = generate_json.get('BuyerDtls', {}).get('GSTIN')
+                    received_total_invoice_value = decoded_response.get('ValDtls', {}).get('TotInvVal')
+                    expected_total_invoice_value = generate_json.get('ValDtls', {}).get('TotInvVal')
+                    # Check for mismatch between decoded and expected e-invoice details:
+                    # - Buyer GSTIN must match
+                    # - Total Invoice Value must be within the allowed government tolerance range:
+                    #   For example, if the expected invoice value is 100,
+                    #   the valid range is from 99 (value - 1) to 101 (value + 1),
+                    #   i.e., 99.00 < received value < 101.00
+                    if (received_gstin != expected_gstin or
+                        not expected_total_invoice_value - 1 < received_total_invoice_value < expected_total_invoice_value + 1
```

Comment:
Do not you actually know the exact amount as you were supposed to send it yourself or is the government secretly doing some rounding or is there a case where we would use this for import?  (I am wondering)

Reply:
some time user go to portal and manually create it so we do this